planner/core/find_best_task.go

```diff
@@ -1032,6 +1032,15 @@ func (ds *DataSource) convertToIndexScan(prop *property.PhysicalProperty, candid
 	task = cop
 	if cop.tablePlan != nil && ds.tableInfo.IsCommonHandle {
 		cop.commonHandleCols = ds.commonHandleCols
+		commonHandle := ds.handleCols.(*CommonHandleCols)
+		for _, col := range commonHandle.columns {
```

Comment:
Would this logic also be implemented in `func (p *LogicalJoin) constructInnerIndexScanTask`?

Reply:
Yes. I constructed a SQL which would fail to run without this logic.

Reply:
added this logic into `constructInnerIndexScanTask` and added the case into unit tests.